pkg/util/util.go

```diff
@@ -147,3 +152,66 @@ func EnsureValidName(name string) string {
 		return c
 	}, name)
 }
+
+func DeleteXfrmRules() error {
+	netLink := netlink.New()
+
+	currentXfrmPolicyList, err := netLink.XfrmPolicyList(syscall.AF_INET)
+	if err != nil {
+		return errors.Wrap(err, "error retrieving current xfrm policies")
+	}
+
+	if len(currentXfrmPolicyList) > 0 {
+		klog.Infof("Cleaning up %d XFRM policies", len(currentXfrmPolicyList))
+	}
+
+	for i := range currentXfrmPolicyList {
+		// These xfrm rules are not programmed by Submariner, skip them.
+		if currentXfrmPolicyList[i].Dst.String() == "0.0.0.0/0" &&
+			currentXfrmPolicyList[i].Src.String() == "0.0.0.0/0" && currentXfrmPolicyList[i].Proto == 0 {
+			klog.V(log.DEBUG).Infof("Skipping deletion of XFRM policy %s", currentXfrmPolicyList[i])
+			continue
+		}
+
+		klog.V(log.DEBUG).Infof("Deleting XFRM policy %s", currentXfrmPolicyList[i])
+
+		if err = netLink.XfrmPolicyDel(&currentXfrmPolicyList[i]); err != nil {
+			return errors.Wrapf(err, "error deleting XFRM policy %s", currentXfrmPolicyList[i])
+		}
+	}
+
+	return nil
+}
+
+func DeleteIfaceAlongWithRoutes(iface string, tableID int) error {
+	link, err := nl.LinkByName(iface)
```

Comment:
Instead of using the `vishvananda` interface directly, we can use our `netlink.Interface` abstraction here and in the function calls below.

Reply:
Done